src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java

```diff
@@ -294,7 +409,24 @@ public void checkCoupling() {
          */
         private boolean isSignificant(String candidateClassName) {
             return !excludedClasses.contains(candidateClassName)
-                    && !candidateClassName.startsWith("java.lang.");
+                    && !isFromExcludedPackage(candidateClassName);
+        }
+
+        /**
+         * Checks if given class should be ignored as it belongs to excluded package.
+         * @param candidateClassName class to check
+         * @return true if we should not count this class.
+         */
+        private boolean isFromExcludedPackage(String candidateClassName) {
+            String classNameWithPackage = candidateClassName;
+            if (!candidateClassName.contains(DOT)) {
+                classNameWithPackage = parentContext
+                    .getClassNameWithPackage(candidateClassName)
+                    .orElse(null);
+            }
+
+            return classNameWithPackage != null && excludedPackages.stream()
+                .anyMatch(classNameWithPackage::startsWith);
```

Comment:
it is better to provide user fine grained ability to define what packages to exclude.
Original request was written a long time ago, author might not considered such cases at all.
Lets for now have strict match algorithm.
We could extend it to list of RegExps in future if user show interest to this.